spec/system/admin/payment_method_spec.rb

```diff
@@ -335,4 +335,62 @@
       end
     end
   end
+
+  context "when updating a payment method with invalid data" do
+    let(:payment_method) { create(:payment_method, :flat_rate, amount: 10) }
+
+    before do
+      Spree::Preferences::Store.instance.persistence = true
+      login_as_admin_and_visit spree.edit_admin_payment_method_path payment_method
+      fill_in "payment_method_name", with: ""
+      fill_in "payment_method_description", with: "Edited description"
+      uncheck "payment_method_distributor_ids_#{@distributors[0].id}"
+      fill_in "Amount", with: 'invalid'
+      click_button 'Update'
+    end
+
+    after do
+      Spree::Preferences::Store.instance.persistence = false
+    end
```

Comment:
Database persistence is necessary to test that when the cache is cleared, it will display the data fetched from the database upon revisiting the edit page. I will add a comment about that to the spec.
Also, I ran the test with the ```around``` block and unfortunately it didn't pass because there was no database persistence. I'm not certain, but it may have something to do with [examples in the around block are run with the configured context](https://relishapp.com/rspec/rspec-core/v/3-12/docs/hooks/around-hooks#examples-run-by-an-%60around%60-block-are-run-in-the-configured-context).

Reply:
Interesting, thanks for trying it.